hcloud/instances.go

```diff
@@ -34,84 +34,100 @@ func newInstances(client *hcloud.Client) *instances {
 	return &instances{client}
 }
 
-func (i *instances) NodeAddressesByProviderID(providerID string) ([]v1.NodeAddress, error) {
+func (i *instances) NodeAddressesByProviderID(ctx context.Context, providerID string) ([]v1.NodeAddress, error) {
 	id, err := providerIDToServerID(providerID)
 	if err != nil {
 		return nil, err
 	}
 
-	server, err := getServerByID(i.client, id)
+	server, err := getServerByID(i.client, id, ctx)
```

Comment:
FYI: It’s a generally-accepted convention to pass `ctx` always as the first argument.